docs/hugo/content/contributing/add-a-new-code-generated-resource/run-the-code-generator.md

```diff
@@ -0,0 +1,224 @@
+```
+
+For example, taking the _Azure Synapse Workspace_ sample from above:
+
+``` yaml
+synapse:
+ 2021-06-01:
+   Workspace:
+     $exportAs: Workspace
+     $supportedFrom: v2.0.0
+```
+
+**Tip:** The `$exportAs` directive allows the name of the resource to be changed. We typically use this to simplify the name, relying on the **group** to disambiguate things. For example, if the original resource name was `SynapseWorkspace` we'd prefer to use `Workspace` to avoid stuttering.
+
+If ASO was already configured to generate resources from this group (or version), you will need to add your new
+configuration around the existing values.
+
+## Run the code generator
+
+In your development environment, run the code generator:
+
+```bash
+task 
+```
+
+If you want to run the code generator manually, work from the `v2` folder and first build the generator yourself:
+
+``` bash
+cd v2
+pushd ./tools/generator; go build -o aso-gen; popd
+```
+
+Then run the generator:
+
+``` bash
+./tools/generator/aso-gen gen-types azure-arm.yaml
+```
+
+## Fix any errors raised by the code generator
+
+It's very likely your first run of the code generator will fail with some verbose errors. 
+
+_**These errors are normal.**_
+
+We have built the code generator to catch most of the common problems that occur when adding a new resource, and the errors are designed to be as comprehensive as possible and to guide you through the process of fixing them.
+
+The most common errors are listed below, but if you need help with a specific error, feel free to contact us either by [creating an issue](https://github.com/Azure/azure-service-operator/issues) or by joining us over on the [Kubernetes Slack](https://kubernetes.slack.com) in the [#azure-service-operator](https://kubernetes.slack.com/archives/C046DEVLAQM) channel
+
+### \<Resource\> looks like a resource reference but was not labelled as one
+
+We have heuristics to identify properties that look like they may be ARM References, but where they aren't explicitly identified as such in the OpenAPI spec.
+
+For each of these errors, you need to add configuration to tell the code generator whether the property is an ARM ID or not. 
+
+Example:
+``` 
+Replace cross-resource references with genruntime.ResourceReference:  
+["github.com/Azure/azure-service-operator/v2/apis/containerservice/v1api20210501/PrivateLinkResource.Id" 
+looks like a resource reference but was not labelled as one.
+```
+
+To fix this, determine whether the property in question is an ARM ID or not, and then update the `objectModelConfiguration` section in the configuration file, as follows:
+
+* Find the section you added earlier
+* Add a new object if it's not already present
+* Adding your property with an `$referenceType:` declaration nested below.
+
+If the property is an ARM ID, use `$referenceType: arm` to flag that property as a reference:
+
+```yaml
+network:
+  2020-11-01:
+    NetworkSecurityGroup:
+      $export: true
```

Comment:
Is this the same configuration file referenced above? There are different properties listed here (no exportas, supportfrom and has export) which might be confusing.